mongo/mongo.go

```diff
@@ -83,3 +82,16 @@ func (m *Mongo) IsConnected(ctx context.Context) bool {
 	err := m.Connection.Ping(ctx, 5)
 	return err == nil
 }
+
+// GetCacheTime returns a cache time with its given id
+func (m *Mongo) GetCacheTime(ctx context.Context, id string) (*models.CacheTime, error) {
+	filter := bson.M{"_id": id}
+	var result models.CacheTime
+
+	err := m.Connection.Collection(m.ActualCollectionName(config.DatasetsCollection)).FindOne(ctx, filter, &result)
+	if err != nil {
+		log.Error(ctx, "Error finding dataset:", err)
```

Comment:
This one is tricky: Depending on the type of error, this may not be the right thing to do. For instance, once we implement the 404 ticket, I believe that when an item is not found, an `ErrNoDocumentFound` error is returned, which is a perfectly valid use case. Maybe we need a comment here explaining all this.

Reply:
Yeah, so currently if this conditional is hit - it is a returning an internal server error. As we connect to the db when we start the app, I'm wondering the use case for an internal server error. Perhaps it would always just be a 404 not found? In any case should it be handled as a part of the next ticket?

Reply:
We can handle it as part of the 404 ticket, but that's why I suggested leaving a message here. So what I meant is that this `err` will not always be an actual server error (if I've understood this correctly). If the type of the error is a `ErrNoDocumentFound`, then that's a valid use case for us (to be handled in the 404 ticket). So as part of that ticket, I think we need to differentiate between both. I was suggesting to leave a comment here explaining that, so that we don't forget.

Does that make sense?

Reply:
Yes that makes sense, added a comment as a reminder.